libraries/resource_providers/api200/logical_interconnect_provider.rb

```diff
@@ -129,6 +149,10 @@ def update_ethernet_settings
       end
 
       def update_port_monitor
+        @item.retrieve! || raise("#{@resource_name} '#{@name}' not found!")
+        ports = convert_keys(@new_resource.port_monitor, :to_s)
```

Comment:
likely better included inside `load_port`

Reply:
Best not, because ``` loadPorts``` will only be executed if the attribute ```enablePortMonitor``` is true. I don't think "ports" is a good name. I think I'd better to change to  "port_monitor". What do you think?

Reply:
I believe it to be more costly outside than inside the `load_port` method, you'd just need to change:
```ruby
load_port(ports) if ports['enablePortMonitor']
```
to
```ruby
load_port(ports) if @new_resource.port_monitor.any?
```
or something similar

Reply:
As for the name of the variable, both `ports` and `port_monitor` are somewhat abstract to me.
Perhaps `port_monitor_attributes`? Or something similar. 
Just suggestions, not a particular issue to me with the naming at least.